src/components/AdvertisingProvider.js

```diff
@@ -6,87 +6,96 @@ import AdvertisingContext from '../AdvertisingContext';
 import equal from 'fast-deep-equal';
 
 export default class AdvertisingProvider extends Component {
-    constructor(props) {
-        super(props);
-        this.initialize();
+  constructor(props) {
+    super(props);
+    this.initialize();
+
+    this.state = {
+      activate: this.advertising.activate.bind(this.advertising),
+    };
+  }
 
-        this.state = {
-            activate: this.advertising.activate.bind(this.advertising),
-        };
+  componentDidMount() {
+    if (this.advertising.isConfigReady() && this.props.active) {
+      this.advertising.setup();
     }
+  }
 
-    componentDidMount() {
-        if (this.advertising.isConfigReady() && this.props.active) {
-            this.advertising.setup();
-        }
+  async componentDidUpdate(prevProps) {
```

Comment:
This is the actual bugfix: we got a lot of “null is not an object” because of line 31. The `teardown` method sets `this.advertising` to `null`.